tzatziki-spring/src/main/java/com/decathlon/tzatziki/steps/SpringSteps.java

```diff
@@ -105,6 +115,13 @@ public void theCacheWillContain(Guard guard, String cacheName, Object message) {
         guard.in(objects, () -> Mapper.<Map<String, Object>>read(this.objects.resolve(message)).forEach(cache::put));
     }
 
+    @After(order = 10001)
+    public void after() {
+        if(clearThreadPoolExecutor && taskExecutor != null) {
+            taskExecutor.shutdown();
```

Comment:
I think we should add an awaitTermination call to ensure all remaining tasks are ended : https://stackoverflow.com/a/18426020

Reply:
Spring uses https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/scheduling/concurrent/ExecutorConfigurationSupport.html

By default **setWaitForTasksToCompleteOnShutdown** is false so by default all remaining tasks are ended